docs/decisions/0003-token-exam-access.rst

```diff
@@ -0,0 +1,50 @@
+Token System for Exam Access
+=================================
+
+Status
+------
+
+Approved (circa September 2022)
+
+Context
+-------
+Exam integrity and security is a fundamental element of special exams on the edX platform, which includes
+timed exams, practice proctored exams, onboarding exams, and proctored exams. In order to ensure exam
+integrity and security, edX learners should only be able to see special exam content when they are authorized to do so.
+
+In the legacy experience for exams, ``edx-proctoring``, exam access was gated by frontend interstitials, as well as a
+check on the backend before rendering the student view. The access check on the backend was possible because the LMS
+directly imports ``edx-proctoring``, and has access to specific functions that return whether or not the content
+should be visible to a learner in an exam. More details on the legacy experience can be found in the `original discovery
+doc <https://2u-internal.atlassian.net/wiki/spaces/PT/pages/15440845/MST-1210+How+do+we+do+access+control+for+exam+content+without+edx-proctoring>`_.
+
+In the new experience, which uses the ``edx-exams`` IDA as a backend, the LMS cannot directly import any functionality from
+``edx-exams``, as it is an idea. Therefore, while the frontend gating still exists via the same interstitials from the legacy
+experience, there is no way to gate access to exam content on the backend, meaning that the content could be
+directly accessed via URL.
+
+With the new experience using ``edx-exams`` as a backend, we need a way to maintain access control.
+
+Decision
+--------
+A JWT token will be used to determine if a user has access to view an x-block. The LMS will not make any direct
+calls to the ``edx-exams`` service to determine if a user has access to view the content of an exam.
```

Comment:
Technically, the LMS is making the call to edx-exams and edx-exams will make the call to token-utils. But I'm guessing this is fine/still not considered a direct call.

Reply:
By LMS I mean edx-platform (i.e. the backend) which shouldn't be making any direct calls to edx-exams for access control. The frontend does though.

Reply:
Ah, got it! Makes sense.